cl/citations/match_citations.py

```diff
@@ -169,116 +186,82 @@ def filter_by_matching_antecedent(
-            # refinement may still fail (because the guess could be
-            # meaningless), in which case the citation is not resolvable and
-            # is dropped.
-            candidates = []
-            for cm in citation_matches:
-                for c in cm.cluster.citations.all():
-                    if (
-                        citation.reporter == c.reporter
-                        and citation.volume == str(c.volume)
-                    ):
-                        candidates.append(cm)
-
-            candidates = list(set(candidates))  # Remove duplicate matches
-            if len(candidates) == 1:
-                # Accept the match!
-                matched_opinion = candidates[0]
-            else:
-                matched_opinion = filter_by_matching_antecedent(
-                    candidates, citation.antecedent_guess
-                )
-
-        # Otherwise, the citation is just a regular citation, so try to match
-        # it directly to an opinion
-        elif isinstance(citation, FullCaseCitation):
-            matches = match_citation(citation, citing_doc=citing_opinion)
-
-            if len(matches) == 1:
-                match_id = matches[0]["id"]
-                try:
-                    matched_opinion = Opinion.objects.get(pk=match_id)
-                except Opinion.DoesNotExist:
-                    # No Opinions returned. Press on.
-                    pass
-                except Opinion.MultipleObjectsReturned:
-                    # Multiple Opinions returned. Press on.
-                    pass
-            else:
-                # No match found for citation
-                pass
-
-        # If an opinion was successfully matched, add it to the list and
-        # set the match fields on the original citation object so that they
-        # can later be used for generating inline html
-        if matched_opinion:
-            was_matched = True
-            citation_matches.append(matched_opinion)
-            citation.match_url = matched_opinion.cluster.get_absolute_url()
-            citation.match_id = matched_opinion.pk
-        else:
-            was_matched = False
-
-    return citation_matches
+    # Only accept if one candidate remains
+    if len(candidates) == 1:
+        return candidates[0]
+
+    # Otherwise, try to refine further using the antecedent guess
+    else:
+        return filter_by_matching_antecedent(
+            candidates, short_citation.antecedent_guess
+        )
+
+
+def resolve_supra_citation(
+    supra_citation: SupraCitation,
+    resolved_full_cites: Dict[FullCaseCitation, Opinion],
+) -> Optional[Opinion]:
+    return filter_by_matching_antecedent(
+        resolved_full_cites.values(), supra_citation.antecedent_guess
+    )
+
+
+def do_resolve_citations(
+    citations: List[CitationBase], citing_opinion: Opinion
+) -> Dict[Union[Opinion, Resource], List[SupportedCitationType]]:
+    # Set the citing opinion on FullCaseCitation objects for later matching
```

Comment:
We need a reference to the citing opinion to optimize our matching of full citations. (Previously this was the `citing_doc` parameter.)